src/components/user/UserProfile.js

```diff
@@ -10,10 +10,21 @@ import {
 import { Avatar } from '@mui/material';
 import { Link } from 'react-router-dom';
 import { useAuth } from '@akosasante/react-auth-context';
+import axiosInstance from '../../apis/axiosClient';
 
 export default function PersonalProfile() {
   const { user } = useAuth();
   const userData = user.user; //the user that's returned is nested in its original response shape so to use the actual user you'll need to unwrap it
+  const [purchasedAlbums, setPurchasedAlbums] = useState([]);
+
+  useEffect(() => {
+    const fetchPurchasedAlbums = async () => {
+      const response = await axiosInstance.get('/users/showMe/withAlbums');
+      console.log(response.data.user.purchasedAlbums);
+      setPurchasedAlbums(response.data.user.purchasedAlbums);
+    };
+    fetchPurchasedAlbums();
+  }, [setPurchasedAlbums]);
```

Comment:
![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/fe48ff0e-4ad6-40e0-b229-2ff7784537c1)

On my user, I have a lot of duplicate albums, because I have purchased the same album lots of times 😅. I'm guessing that should be allowed in case I want multiple copies of the album for whatever reason as a customer. But when showing them here, maybe we should do a `purchasedAlbums.filter` here first to remove any duplicates (or use a helper function from a library like [lodash](https://stackoverflow.com/questions/31740155/lodash-remove-duplicates-from-array) or `underscore`) just to clean up the UI a bit.